x/evm/keeper/state_transition.go

```diff
@@ -347,7 +346,7 @@ func (k *Keeper) ApplyMessageWithConfig(ctx sdk.Context,
 		// NOTE: this only adds active precompiles to the EVM.
 		// This means that evm.Precompile(addr) will return false for inactive precompiles
 		// even though this is actually a reserved address.
```

Comment:
ideally we can also tackle the problem noted in this comment when we are looking at refactoring this?